feat: new theme for ECharts 6.0

Ovilia opened the pull request:
## Brief Information

This pull request is in the type of:

- [ ] bug fixing
- [x] new feature
- [ ] others

### What does this PR do?

This PR proposes a new default theme for Apache ECharts 6.0.

## Design Concepts

Key Words: **Modern, Confident, Versatile**

### Challenges

We have observed that many Apache ECharts users use the default theme in their actual products, which presents a significant challenge for us to provide an excellent default theme. When designing the new theme for 6.0, I aimed to address the following issues:

1. Component overlap: Since ECharts components (chart grid area, legend, title, etc.) use absolute positioning and components cannot easily share position information, components often overlap in many scenarios
2. Inconsistent design language: Existing code often uses literal color and distance values, resulting in arbitrary use of colors (especially neutral colors), distances, shadows, and icons across different components, lacking the aesthetic appeal of a cohesive design
3. Aesthetic issues: Some components have too little whitespace, making the layout feel crowded and too close to the edges, lacking a sense of psychological security

### Solution

Therefore, I have defined color and distance tokens uniformly in the `tokens.ts` file. By referencing these variables, we can ensure design consistency and improve code readability. Then, I refined the spacing and positioning of each component and chart to ensure elegant display under different chart types and series quantities.

In terms of theme color design, I chose colors with higher value (brightness), injecting more modern vitality into the charts, conveying a confident and clear visual language that makes data expression more attractive. The new color scheme is bold and innovative while maintaining continuity - the first two theme colors (blue and green) retain their hue, preserving familiar visual anchors while injecting freshness into the overall style.

## Before & After

In `test/theme-list.html`, I have provided examples of typical chart types and components with series quantities ranging from 1 to 9. Here, we will explain some typical charts.

### Typical Charts

Let's start by experiencing the new design with common 4-series bar and pie charts.

|ECharts 5|ECharts 6|
|-|-|
|<img width="506" alt="image" src="https://github.com/user-attachments/assets/ac0e3f8c-7a37-4ef7-bfb5-d5ea28678eab" /><img width="505" alt="image" src="https://github.com/user-attachments/assets/9321da29-1314-4cd9-a914-5e464674805b" />|<img width="505" alt="image" src="https://github.com/user-attachments/assets/d9032180-0f54-48de-993f-e06fdb1a3be5" /><img width="505" alt="image" src="https://github.com/user-attachments/assets/77157060-5711-4ff1-9aed-ef98caae2ec1" />|

In ECharts 5, the `title` and `legend` often overlap. Before we come up with a better layout plan, I moved the `legend` to the bottom, not only avoiding overlap but also creating a more balanced layout visually with the `title`. We left more space at the bottom of the `grid` to prevent overlap even with multiple legend items, and enabled `containLabel` by default to ensure axis labels don't exceed the canvas when data values are large.

|ECharts 5|ECharts 6|
|-|-|
|<img width="505" alt="image" src="https://github.com/user-attachments/assets/47e6ff53-89f8-46ba-a4b6-73256ef84edf" /><img width="505" alt="image" src="https://github.com/user-attachments/assets/cf482c9f-f11e-41a6-888d-26d094975b5b" />|<img width="505" alt="image" src="https://github.com/user-attachments/assets/fbd85271-63db-4ead-91d8-87ce822d57f8" /><img width="506" alt="image" src="https://github.com/user-attachments/assets/e06d42ba-5fe1-4284-8e84-f63de98d553c" />|

### Theme Colors

To handle different chart usage scenarios, we need to ensure the theme colors look harmonious with different numbers of series. Below, using bar charts as an example, we show the effects with 1-9 series, and then explain how we balance the design as series number varies.

|ECharts 5|ECharts 6|
|-|-|
|<img width="505" alt="image" src="https://github.com/user-attachments/assets/81b071b2-66ea-4469-90cb-aebc2a15ca76" /><img width="505" alt="image" src="https://github.com/user-attachments/assets/b040e8a3-63a3-4f19-bab7-7e143bb26d91" /><img width="504" alt="image" src="https://github.com/user-attachments/assets/c5ec20de-837d-4d6e-a4c4-4f7b55db7a78" /><img width="505" alt="image" src="https://github.com/user-attachments/assets/46d62e7b-2eb2-4d79-9188-069e34567265" /><img width="504" alt="image" src="https://github.com/user-attachments/assets/0d6037ae-e812-47f7-b642-8f5312588b6a" /><img width="504" alt="image" src="https://github.com/user-attachments/assets/f95511a8-8070-42e5-8567-101fc35ad145" /><img width="505" alt="image" src="https://github.com/user-attachments/assets/d2c99b48-c3a2-421e-bfd9-725aefbe3c63" /><img width="505" alt="image" src="https://github.com/user-attachments/assets/162d8f56-d13b-4745-b4c5-eef01b6a3500" /><img width="505" alt="image" src="https://github.com/user-attachments/assets/fa50c3ef-46e1-4708-80f2-e887a3fb342d" />|<img width="505" alt="image" src="https://github.com/user-attachments/assets/cce8843c-b2ae-47a2-ba88-893684df108a" /><img width="505" alt="image" src="https://github.com/user-attachments/assets/862beae5-0506-4c3f-a7fa-4b319595adc5" /><img width="505" alt="image" src="https://github.com/user-attachments/assets/6e688778-ad70-4331-be18-2ef8db65490b" /><img width="504" alt="image" src="https://github.com/user-attachments/assets/6d5dc416-0118-417a-8bcc-03ffbae20d98" /><img width="505" alt="image" src="https://github.com/user-attachments/assets/97510c70-3220-4486-8fb8-34465434d6f4" /><img width="505" alt="image" src="https://github.com/user-attachments/assets/5335701b-2ef7-40ad-88f9-88fd74dbd322" /><img width="506" alt="image" src="https://github.com/user-attachments/assets/cfca4df2-dde9-4ebc-96ca-9fb5518aaac0" /><img width="505" alt="image" src="https://github.com/user-attachments/assets/0dc238a8-375c-4b7b-8533-6480d1fcadf0" /><img width="505" alt="image" src="https://github.com/user-attachments/assets/4cc8eafc-f0e1-419e-8fac-e2f6a161d067" />|

#### How to Ensure Theme Color Harmony

The biggest challenge in designing theme colors is how to ensure distinction while maintaining overall harmony when there are many series?

##### Warm-Cool Consistency

This inspiration came from a painting course where I learned this technique: when multiple colors are needed in a composition, overall harmony can be achieved by unifying the warm-cold nature of the colors.

<img width="833" alt="image" src="https://github.com/user-attachments/assets/8427fe5e-9641-4cc7-a1f5-1f007a3029d9" />

In the image above, we can see that colors with consistent warm-cold tones appear more harmonious when grouped together. In the theme colors I propose, I use colors that lean towards cool tones because cool colors give a sense of professionalism and trustworthiness, which is more suitable for most chart scenarios.

##### Value and Lightness Variation

If we carefully observe the three-series bar chart, we can see that the color scheme in 6 is more harmonious than 5. Why is that?

|ECharts 5|ECharts 6|
|-|-|
|<img width="504" alt="image" src="https://github.com/user-attachments/assets/c5ec20de-837d-4d6e-a4c4-4f7b55db7a78" />|<img width="505" alt="image" src="https://github.com/user-attachments/assets/6e688778-ad70-4331-be18-2ef8db65490b" />|

After converting to grayscale, we can see the reason more clearly.

|ECharts 5|ECharts 6|
|-|-|
|<img width="505" alt="image" src="https://github.com/user-attachments/assets/4b7ae952-8f76-4f90-a9db-35f5e350edb8" />|<img width="505" alt="image" src="https://github.com/user-attachments/assets/64480557-6e9c-43c2-a219-5089f2058761" />|

When lightness follows a monotonically linear progression (as seen in ECharts 5's first three series), it introduces visual instability. By implementing dynamic lightness modulation through non-linear curves, we achieve both enhanced accessibility and harmonious aesthetics.

|ECharts 5|ECharts 6|
|-|-|
|<img width="505" alt="image" src="https://github.com/user-attachments/assets/00fd1e17-c4dc-486d-8159-21ee801921e5" />|<img width="505" alt="image" src="https://github.com/user-attachments/assets/7910ba62-1f31-4293-9093-12ecb94fc1c1" />|

Besides lightness, value also needs this kind of variation, so the third theme color uses a lower value to make the overall look more balanced. If the third theme color used a similar value (like the orange we ultimately placed in the fourth position), although it wouldn't feel out of place, I think it would lack some sophistication from an aesthetic perspective, as it would be like a tripod lacking focus, making it an unstable design.

|If the third theme color used similar value, although acceptable, lacks sophistication|The third theme color actually chosen in this PR|
|-|-|
|<img width="505" alt="image" src="https://github.com/user-attachments/assets/f8d12fc8-058a-4426-8ae0-cee8064d8ef8" />|<img width="505" alt="image" src="https://github.com/user-attachments/assets/6e688778-ad70-4331-be18-2ef8db65490b" />|

### Common Charts and Components

Due to space limitations, we cannot show the effects of different series quantities for various charts here, only showing the effects with four series as an example. The complete examples can be found in the zip file attached. [echarts-themes-v5-vs-v6.zip](https://github.com/user-attachments/files/19905771/echarts-themes-v5-vs-v6.zip)

#### Bar Chart

|ECharts 5|ECharts 6|
|-|-|
|<img width="504" alt="image" src="https://github.com/user-attachments/assets/c5ec20de-837d-4d6e-a4c4-4f7b55db7a78" />|<img width="505" alt="image" src="https://github.com/user-attachments/assets/6e688778-ad70-4331-be18-2ef8db65490b" />|

Hid `axisTick` for 'category' axis because in most cases, the distance relationship already indicates grouping information, and `axisTick` is not needed for precise division. Removing it makes the visual more continuous and reduces unnecessary redundant information. More space is left at the bottom of the grid to prevent overlap when the legend has two rows.

#### Line Chart

|ECharts 5|ECharts 6|
|-|-|
|<img width="505" alt="image" src="https://github.com/user-attachments/assets/1e239ddc-bc81-4600-9b6d-89a0c4cf4a0a" />|<img width="505" alt="image" src="https://github.com/user-attachments/assets/f8d36aa9-4223-45a3-9db1-831f1fd0010b" />|

Increased `lineStyle.width` and `symbolSize`, making the overall look more modern and confident. The original `symbolSize` was too close to the line width, lacking distinction, thus not aesthetically pleasing.

#### Pie Chart

|ECharts 5|ECharts 6|
|-|-|
|<img width="505" alt="image" src="https://github.com/user-attachments/assets/9321da29-1314-4cd9-a914-5e464674805b" />|<img width="505" alt="image" src="https://github.com/user-attachments/assets/77157060-5711-4ff1-9aed-ef98caae2ec1" />|

Reduced `radius` to add more whitespace for aesthetic appeal and avoid overlap with other components. Increased `labelLine` width.

#### Scatter Chart

|ECharts 5|ECharts 6|
|-|-|
|<img width="505" alt="image" src="https://github.com/user-attachments/assets/49284596-f2ed-4d0f-8f3a-e2987b712eb5" />|<img width="504" alt="image" src="https://github.com/user-attachments/assets/41d80622-400d-40bb-9ece-7a994798e682" />|

#### Funnel Chart

|ECharts 5|ECharts 6|
|-|-|
|<img width="505" alt="image" src="https://github.com/user-attachments/assets/e26952ee-d65a-4682-b8ad-aa08bea81222" />|<img width="504" alt="image" src="https://github.com/user-attachments/assets/36df96a6-28a5-4a29-9da6-928f5fde1546" />|

#### Treemap Chart

|ECharts 5|ECharts 6|
|-|-|
|<img width="503" alt="image" src="https://github.com/user-attachments/assets/e9574f9b-dede-4fe3-9f1f-ad7ae56e6d03" /><img width="456" alt="image" src="https://github.com/user-attachments/assets/4bf9d858-1367-4e99-af66-a0cbb822ce06" />|<img width="504" alt="image" src="https://github.com/user-attachments/assets/220025e3-d4b7-4415-b329-f9ef22b7e926" /><img width="291" alt="image" src="https://github.com/user-attachments/assets/cdecd41a-7e96-43a2-9957-05c8c92d5b00" />|

The original layout used percentage positioning on the left and right, which was changed to fixed pixel size, and this size is the same as the margin used by other components (such as heatmap controls, Toolbox, etc.). The original `breadcrumb` dark gray background was visually too heavy, but logically it's not the main information, so it was changed to a light gray background.

#### Sunburst Chart

|ECharts 5|ECharts 6|
|-|-|
|<img width="505" alt="image" src="https://github.com/user-attachments/assets/e2cd3601-d546-4a61-9b1a-739906c7f090" />|<img width="505" alt="image" src="https://github.com/user-attachments/assets/868f3c94-8cfd-4ee2-8dff-dd2cecc10b6a" />|

|After Drilling Down, ECharts 5|After Drilling Down, ECharts 6|
|-|-|
|<img width="505" alt="image" src="https://github.com/user-attachments/assets/2ef93c9f-20d8-404d-9728-2c334f3b66e7" />|<img width="505" alt="image" src="https://github.com/user-attachments/assets/e8990be2-1d19-464f-a30e-d67e333af460" />|

Originally, the first theme color was used for the drill-down return node, causing the default view to start from the second theme color, making it look different from other charts. Therefore, it was changed to use gray for drill-down nodes.

#### ThemeRiver Chart

|ECharts 5|ECharts 6|
|-|-|
|<img width="504" alt="image" src="https://github.com/user-attachments/assets/32fe6ded-c3de-4c59-81ec-faf0cf9b3dfd" />|<img width="505" alt="image" src="https://github.com/user-attachments/assets/8c9616ba-93bf-4840-86b0-a3df36440807" />|

Random data so it doesn't look very similar, but this chart actually didn't have significant changes.

#### Timeline

|ECharts 5|ECharts 6|
|-|-|
|<img width="505" alt="image" src="https://github.com/user-attachments/assets/66fd25ab-ee24-4fe9-8639-8d86cf0d1dbd" />|<img width="505" alt="image" src="https://github.com/user-attachments/assets/efbb8e9f-312a-422a-bfd4-71a12cf737a3" />|

Adjusted colors to avoid using gray which might be mistaken for disabled state. Redrew icon SVGs to maintain width consistency. Removed strokes and shadows because ECharts' overall design is flat, and shadows should be avoided when they don't help prevent misunderstanding.

#### DataZoom & Toolbox

|ECharts 5|ECharts 6|
|-|-|
|<img width="504" alt="image" src="https://github.com/user-attachments/assets/2f904448-b2ed-4392-a717-ac2b010024e1" />|<img width="505" alt="image" src="https://github.com/user-attachments/assets/48158d71-893c-4310-a93d-93298f352b15" />|

Adjusted colors to avoid using gray which might be mistaken for disabled state.

#### VisualMap & Calendar

|ECharts 5|ECharts 6|
|-|-|
|<img width="505" alt="image" src="https://github.com/user-attachments/assets/a2c406ed-0a5a-4d78-b85f-2275bee7e241" />|<img width="505" alt="image" src="https://github.com/user-attachments/assets/5f335b7a-4263-4c19-b9f7-12d725472fef" />|

ECharts 5 used a red to yellow gradient, but this color only appeared in VisualMap, lacking connection with the overall design. It was adjusted to a gradient related to the first theme color to make it more consistent with other charts and avoid the warning implications of red, since larger values don't always indicate better or worse. The VisualMap control has more whitespace on the sides, with this distance matching the margins of other components. Slightly adjusted Calendar colors.

#### Map

|ECharts 5|ECharts 6|
|-|-|
|<img width="505" alt="image" src="https://github.com/user-attachments/assets/3a2ae38e-5248-48c3-8760-ffb6ea5f4f8a" /><img width="504" alt="image" src="https://github.com/user-attachments/assets/3f640e21-bbec-4652-b621-4525f8a626a5" />|<img width="505" alt="image" src="https://github.com/user-attachments/assets/8ff7161d-f3f5-429f-8f63-238bb502038a" /><img width="504" alt="image" src="https://github.com/user-attachments/assets/b3beadf1-e8a2-4d27-8386-35a80cd22b09" />|

The yellow highlight background in ECharts 5 maps wasn't used anywhere else, making it inconsistent with the overall design. Actually, this highlight is similar to the color lifting after emphasis in bar charts, but making it lighter made it hard to see, so it was changed to a slightly darker background color.

### Dark Mode

Examples of some dark mode charts:

<img width="505" alt="image" src="https://github.com/user-attachments/assets/c89cd89f-2124-46f2-bc3b-68ee8e945622" />

<img width="506" alt="image" src="https://github.com/user-attachments/assets/914a7f7b-9b6f-457d-af49-95f6645412bf" />

<img width="506" alt="image" src="https://github.com/user-attachments/assets/bdbf48cb-5a6b-4103-9af0-fb418e21c162" />

To see more cases, toggle system dark mode and check `test/theme-list.html`. (https://github.com/ecomfe/zrender/pull/1118 is required for zrender).

## Document Info

One of the following should be checked.

- [x] This PR doesn't relate to document changes
- [ ] The document should be updated later
- [ ] The document changes have been made in apache/echarts-doc#xxx



## Misc

### ZRender Changes

- [x] This PR depends on ZRender changes (ecomfe/zrender#1118).

### Related test cases or examples to use the new APIs

N.A.



## Others

### Merging options

- [ ] Please squash the commits into a single one when merging.

### Other information

diysimon commented:
Very professional design, looking forward to the official release of 6.0！

Russiarain commented:
Great design! Just one question, will there be enough contrast to the surface(maybe `#1e1e1e`) in dark mode for the third theme color?

Ovilia commented:
@100pah Thanks for the detailed review!

## Backward Compatibility

Since we aim to reduce the overlapping issues among default components, we've had to adjust the default positions and alignment methods of the components. This might cause some situations where users' original settings lead to overlapping under the new layout, just as in the example you mentioned. However, this is **a carefully considered trade-off**. For instance, I believe the probability of displaying both the legend and the title simultaneously is much higher than that of displaying both the legend and the dataZoom. Suppose the new solution can address 60% of the previous bad-case scenarios while introducing 10% of new ones. Even so, this optimization is still highly valuable.

To completely solve this problem, we would need to come up with **a robust component anti-overlapping mechanism**. Unfortunately, we're likely to run out of time to implement this in 6.0. Therefore, when the time comes, we will provide an **migration document** to inform users about which default values we've modified and how they can adjust the positions in case of overlapping.

## Theme Colors

After much careful consideration and deliberation, I truly believe that this new theme is the optimal one I can come up with. When it comes to aesthetics, it might be rather challenging for me to provide objective evidence. However, in my sincere opinion, this is the best design I could come up with. Designing the theme color requires a comprehensive and holistic consideration. It's not simply about having individual colors that look good on their own; they must also harmonize well when put together.

Therefore, **with due respect to all viewpoints, I'm inclined to uphold this theme**. I understand that different people may have diverse perspectives on color aesthetics, but based on my in-depth consideration and analysis, I believe this theme has its unique merits and can best meet our overall design needs. 

I'm more than willing to further discuss and listen to your thoughts to reach a more satisfactory solution together. If any of you have some better ideas regarding the theme color, I would humbly suggest that you could use the [theme builder](https://echarts.apache.org/zh/theme-builder.html) to create a theme think is better. Subsequently, we could further discuss it through a community voting process.

As about the issue that the third color is relatively close to a neutral color, I believe this is a design choice after comprehensively considering various factors. The reasons for this design were also stated in the PR description. If you have any objections, I would suggest you propose a better overall design plan. This PR currently doesn't have the logic for a dark mode. Whether certain colors need to be adjusted may be further discussed in the PR for the dark mode in the future. My preliminary plan is to calculate the colors for the dark mode dynamically based on the light mode. If some colors need adjustment, we can then consider whether corresponding methods need to be provided.

Regarding the concern that the green might be too light and the contrast for labels is insufficient, this issue isn't actually caused by the selection of the theme color. Instead, it stems from the inadequate optimization of label colors for some series. For example, pie and funnel both use label/labelLine design, why is the label color of the pie chart dark gray, while the funnel chart's label takes the theme color? Considering design consistency, could we **change the default label color of the funnel chart to the same dark gray as that of the pie chart**?

In the Treemap, the yellow in the original theme color also has a problem with insufficient brightness contrast between the background color and the label color. Similar to how the [bar chart automatically calculates whether to use white or dark gray for the label](https://echarts.apache.org/examples/zh/editor.html?c=bar-label-rotation) based on the brightness of the theme color when the label is inside, wouldn't it be more reasonable if we adopted the same approach for the Treemap?

## Highlight/emphasis color of map

If you think the current slightly darker gray isn't eye-catching enough, I think we could change it to light blue. However, I'm concerned that this color might cause confusion when using visualMap. Could a darker gray be acceptable to you? The originally yellow and red is never used elsewhere and the color choice is not elegant, so I feel necessary to change them.

I don't quite agree with the statement "since users typically use map individually and do not combinate maps with other types of chart". Maps are likely to appear in the same dashboard as other charts, not to mention that we even support designs like displaying pie charts on the map coordinate.

## Visual map color

Using darker color for smaller value was a mistake. I didn't check carefully before and will fix this later.

Using a mono-tone color in visual mapping could be a better design choice.

Firstly, a mono-tone design is simpler and more likely to achieve harmony. This might well be the reason why GitHub adopts a mono-tone green design.

Secondly, the introduction of red may imply dangerous or negative connotations. However, in many cases, a large value doesn't necessarily carry such meanings. Therefore, for the default design, a more neutral semantic expression is, in my opinion, more appropriate.

Thirdly, in the gradient from red to blue as you suggested, there is a light-blue to dark-blue section. In fact, the value corresponding to light blue is larger than that of dark blue, just as you pointed out above, dark colors usually represent larger values. Thus, using a mono-tone color can reduce such misunderstandings.

## axisTick

As we can see from the following demos, most chart libraries use no axis tick for default.

![Most chart libraries use no axis tick for default](https://github.com/user-attachments/assets/9954ffb0-e577-47a6-81ed-9815fdac79ed)

I believe again, this is a design choice. The benefit of having a neat design by not showing ticks overweighs the disadvantage of data grouping. Maybe we can use a smaller `barGap` by default to better indicate the grouping.

Ticks should not be hidden for time series. Can you provide the option for your time axis for me to check?

Ovilia commented:
In the above commit, I added a new option `grid.containDataZoom`. Similar to `containLabel`, it adjust grid position to make room for dataZoom. @100pah Do you feel this is the correct way to fix the problem?

## v5

<img width="1046" alt="image" src="https://github.com/user-attachments/assets/dd6bb5fe-aeed-424c-86e2-a05ea8b8c679" />

## v6

<img width="1046" alt="image" src="https://github.com/user-attachments/assets/c46df2b2-c71f-4200-b450-433670d60685" />

Option is:

```js
{
  show: true,
  realtime: true,
  start: 30,
  end: 70
},
{
  show: true,
  realtime: true,
  yAxisIndex: 0
}
```

and

```js
{
  show: true,
  realtime: true,
  start: 30,
  end: 70,
  top: 50
},
{
  show: true,
  realtime: true,
  yAxisIndex: 0,
  left: 25
}
```

Ovilia commented:
Updated funnel label style to be compat with pie.

<img width="514" alt="image" src="https://github.com/user-attachments/assets/d7f33d7c-0f37-4b67-aaeb-0cdc87957067" />

Changed map emphasis color to be yellow.

<img width="509" alt="image" src="https://github.com/user-attachments/assets/922bf7ab-7842-4570-879f-09863710a22e" />

Ovilia commented on src/component/tooltip/TooltipModel.ts:
```diff
@@ -137,6 +138,7 @@ class TooltipModel extends ComponentModel<TooltipOption> {
 
         // tooltip border width, unit is px, default is 0 (no border)
         borderWidth: 1,
+        borderColor: tokens.color.borderShade,
```
Yes, this was a bug introduced when I tried to add a tooltip border to dark mode. I'll check this later.

Ovilia commented on src/chart/gauge/GaugeSeries.ts:
```diff
@@ -34,6 +34,7 @@ import {
 } from '../../util/types';
 import GlobalModel from '../../model/Global';
 import SeriesData from '../../data/SeriesData';
+import tokens from '../../visual/tokens';
```
Upated to a darker gray, not exactly the same as v5 because we changed a bit about the neutral colors in v6.

<img width="933" alt="image" src="https://github.com/user-attachments/assets/75aebcb5-da8c-48a5-bd0c-65ce86f2df93" />

Ovilia commented on src/theme/light.ts:
```diff
@@ -1,35 +0,0 @@
-/*
```
Moved to `/theme/rainbow.js`.

Ovilia commented on src/chart/funnel/FunnelSeries.ts:
```diff
@@ -177,12 +178,12 @@ class FunnelSeriesModel extends SeriesModel<FunnelSeriesOption> {
             lineStyle: {
                 // color: 各异,
-                width: 1
+                width: 2
```
Reverted.

Ovilia commented on src/chart/scatter/ScatterSeries.ts:
```diff
@@ -151,7 +152,7 @@ class ScatterSeriesModel extends SeriesModel<ScatterSeriesOption> {
         select: {
             itemStyle: {
-                borderColor: '#212121'
+                borderColor: tokens.color.primary
```
Changed borderWdith to be 2. 
<img width="479" alt="image" src="https://github.com/user-attachments/assets/3aadaf8b-a41b-4769-a041-cf7cefaee026" />

Ovilia commented on src/chart/line/LineSeries.ts:
```diff
@@ -166,7 +167,7 @@ class LineSeriesModel extends SeriesModel<LineSeriesOption> {
 
         lineStyle: {
-            width: 2,
+            width: 3,
```
Reverted.

Ovilia commented on src/coord/calendar/CalendarModel.ts:
```diff
@@ -36,6 +36,7 @@ import {
 } from '../../util/types';
 import GlobalModel from '../../model/Global';
 import Model from '../../model/Model';
+import tokens from '../../visual/tokens';
```
Changed the year label to be a lighter gray, but not as light as v5. I don't quite believe we should make it lighter since it's larger. It's larger because it's more prominant, so it gives more reason to make it darker.

<img width="1150" alt="image" src="https://github.com/user-attachments/assets/8920f0e2-1106-4839-846d-0e1393cd25e5" />

Dark background should not be considered in this case. If user wants to use dark background, he, or the one who makes the dark theme should be responsible for changing the label colors to fit the background. And whatever color you choose for the default label, it's still possible that the user set the background to a similar color (a black background for example).

Ovilia commented on src/component/axisPointer/AxisPointerModel.ts:
```diff
@@ -132,11 +133,7 @@ class AxisPointerModel extends ComponentModel<AxisPointerOption> {
-            shadowColor: '#aaa',
-            shadowOffsetX: 0,
-            shadowOffsetY: 2,
+            color: tokens.color.quaternary,
```
Changed to:
<img width="245" alt="image" src="https://github.com/user-attachments/assets/2191bc83-f27e-4133-a0d4-5b0a0fdafb74" />

Ovilia commented on src/component/legend/ScrollableLegendModel.ts:
```diff
@@ -95,11 +96,11 @@ class ScrollableLegendModel extends LegendModel<ScrollableLegendOption> {
         pageIconSize: 15, // Can be [10, 3], which represents [width, height]
         pageTextStyle: {
-            color: '#333'
+            color: tokens.color.tertiary
```
Changed to:

<img width="86" alt="image" src="https://github.com/user-attachments/assets/b83424f1-efc6-4570-9597-e8e59ef1729b" />

Ovilia commented on src/component/timeline/SliderTimelineModel.ts:
```diff
@@ -99,47 +100,42 @@ class SliderTimelineModel extends TimelineModel {
 
             position: 'left',  // 'left' 'right' 'top' 'bottom'
```
I prefer not to use the shadow because there are no other components using such design language. Changed to:
<img width="290" alt="image" src="https://github.com/user-attachments/assets/dedf2959-2723-44b3-b91e-e060cc1c5d1b" />

100pah commented on src/component/timeline/SliderTimelineModel.ts:
```diff
@@ -99,47 +100,42 @@ class SliderTimelineModel extends TimelineModel {
-                borderColor: '#316BF3',
-                borderWidth: 2
-            }
+                color: tokens.color.accent70
```
I think here it should be :
```ts
            controlStyle: {
                color: tokens.color.accent70,
                borderColor: tokens.color.accent70,
            },
```

Consider users set 
```ts
                            controlStyle: {
                                emphasis: {
                                    borderWidth: 1
                                }
                            },
```
normal style:
<img width="48" alt="image" src="https://github.com/user-attachments/assets/8a65dc5c-024d-4a43-944a-cbde2de6033d" />

emphasis style in this PR (a little bit weird):
<img width="44" alt="image" src="https://github.com/user-attachments/assets/a90c982e-783c-40b9-b1e9-13b014a765f0" />

emphasis style expect:
<img width="52" alt="image" src="https://github.com/user-attachments/assets/b6f050a9-7a5f-4471-8d90-7d768fc38681" />

Ovilia reviewed (commented):
Here is the report of visual tests using current remote `v6` branch and local code with `v5.js` theme file. There are only two imcompatible changes:

1. Sunburst series color

In v5, the first theme color was used for the drill-down return node, causing the default view to start from the second theme color, making it look different from other charts. Therefore, it was changed to use gray for drill-down nodes and series use theme colors starting from the first one instead of the second one.

2. Loading spinner and text color

Since these colors are not using a registered theme, it cannot be override. But the change is acceptable.

All other diffs in the test report are due to the false alarm of the visual test tool and should be ignored.

In the last commit, I implemented a way to support "Mark As Expected" Feature for the test tool so that we can see clearly in the report which diffs are expected.

Here is the full report:
[Visual Regression Test Report-v6-theme.html.zip](https://github.com/user-attachments/files/20518782/Visual.Regression.Test.Report-v6-theme.html.zip)

plainheart commented on src/component/tooltip/TooltipView.ts:
```diff
@@ -57,6 +57,7 @@ import { normalizeTooltipFormatResult } from '../../model/mixin/dataFormat';
 import { createTooltipMarkup, buildTooltipMarkup, TooltipMarkupStyleCreator } from './tooltipMarkup';
 import { findEventDispatcher } from '../../util/event';
 import { clear, createOrUpdate } from '../../util/throttle';
+import tokens from '../../visual/tokens';
```
Remove the unused `tokens` import to fix this lint warning.

Justineo commented:
From my personal point of view, the new color scheme doesn’t feel quite as visually appealing as the previous version. I’m wondering if we might have placed a stronger emphasis on accessibility this time? I’d love to better understand the guiding principles behind how we define the primary color palette — for instance, how accessibility considerations are prioritized in the overall design process. Do we typically validate the results through steps like color vision deficiency simulation or similar regression testing?
